blog/migrate-to-webcontentsview.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Migrating from BrowserView to WebContentsView
+date: 2024-11-11T00:00:00.000Z
+authors: yangannyx
+slug: migrate-to-webcontentsview
+tags: [community]
+---
+
+`BrowserView` has been deprecated since [Electron 30](http://www.electronjs.org/blog/electron-30-0) and is replaced by `WebContentView`. Thankfully, migrating is fairly painless.
+
+---
+
+Electron is moving from [`BrowserView`](https://www.electronjs.org/docs/latest/api/browser-view) to [`WebContentsView`](https://www.electronjs.org/docs/latest/api/web-contents-view) to align with Chromium’s UI framework, the [Views API](https://www.chromium.org/chromium-os/developer-library/guides/views/intro/). `WebContentsView` offers a reusable view directly tied to Chromium’s rendering pipeline, simplifying future upgrades and opening up the possibility for developers to integrate non-web UI elements to their Electron apps. By adopting `WebContentsView`, applications are not only prepared for upcoming updates but also benefit from reduced code complexity and fewer potential bugs in the long run.
+
+Developers familiar with BrowserWindows and BrowserViews should note that `BrowserWindow` and `WebContentsView` are subclasses inheriting from the [`BaseWindow`](https://www.electronjs.org/docs/latest/api/base-window) and [`View`](https://www.electronjs.org/docs/latest/api/view) base classes, respectively. To fully understand the available instance variables and methods, be sure to consult the documentation for these base classes.
+
+## Migration steps
+
+### 1. Upgrade Electron to 30.0.0 or higher
+
+:::warning
+Electron releases may contain breaking changes that affect your application. It’s a good idea to test and land the electron upgrade on your app _first_ before proceeding with the rest of this migration. A list of breaking changes can be found in the release notes for each major version on the Electron Blog.
+:::
+
+### 2. Familiarize yourself with where your application uses BrowserViews
+
+One way to do this is to search your codebase for `new BrowserView(`. This should give you a sense for how your application is using BrowserViews and how many call sites need to be migrated.
+
+:::tip
+For the most part, each instance where your app instantiates new BrowserViews can be migrated in isolation from the others.
+:::
+
+### 3. Migrate each usage of `BrowserView`
+
+1. Migrate the instantiation. This should be fairly straight forward because [WebContentsView](https://www.electronjs.org/docs/latest/api/web-contents-view#new-webcontentsviewoptions) and [BrowserView’s](https://www.electronjs.org/docs/latest/api/browser-view#new-browserviewoptions-experimental-deprecated) constructors have essentially the same shape. Both accept [WebPreferences](https://www.electronjs.org/docs/latest/api/structures/web-preferences) via the `webPreferences` param.
```

Comment:
```suggestion
1. Migrate the instantiation. This should be fairly straightforward because [WebContentsView](https://www.electronjs.org/docs/latest/api/web-contents-view#new-webcontentsviewoptions) and [BrowserView’s](https://www.electronjs.org/docs/latest/api/browser-view#new-browserviewoptions-experimental-deprecated) constructors have essentially the same shape. Both accept [WebPreferences](https://www.electronjs.org/docs/latest/api/structures/web-preferences) via the `webPreferences` param.
```